packages/shared-pure/src/types/Token.ts

```diff
@@ -17,4 +17,5 @@ export interface Token {
   chainId: ChainId
   type: ValueType
   premintHolderAddresses?: EthereumAddress[]
+  iconUrl?: string
```

Comment:
One thing we need to keep in mind, is that conigecko likes to change CoingeckoId of a token from time to time, so probably in such event developer (when updating CoingeckoId) should also check whether the iconUrl still works after the change. Do you have an idea how can we enforce it?

Reply:
We could try to write a test that will go through these iconUrls and try to fetch them. If it's `200` that's good and fail otherwise, but what if Coingecko replaces that url with a placeholder image, then the tests should fail but won't. Ideally we would fetch test image, hash the response and compare the hashes but then we have to store somewhere those hashes. Something to keep in mind for later I suppose.